### ISSUE TITLE
Type checker does not verify symbol existence during assignment

### ISSUE THREAD

**Issue by ElianHugh** [vapourlang/vapour#74]
Hello again, I've been playing around with vapour and ran into a small issue.

Consider the following:

```r
let foo: bar = baz
foo = baz # baz not found
foo = 1 # left expects `bar`, right returns `int`
```

It seems that the checker is inferring that the symbols exist during the assignment. I would expect that the first error would be that the type `bar` does not exist, and then that the value `baz` doesn't exist. However, it appears that this is checked only after variable assignment.

**Comment by JohnCoene** [comment]
Hi Elian,

Thanks so much for all the issues you've opened, it's really helpful! (some are already tackled on branch `v0.0.6`).

Will look into:

- [ ] Check symbols on left hand side of infixes
- [ ] Check that types exist when used

**Comment by JohnCoene** [comment]
Will be fixed in `v0.0.6`, thanks again for sharing!

https://github.com/vapourlang/vapour/blob/7cb944757215c047ea66683ab453d47234dd48e3/walker/walk_test.go#L1359

**Comment by ElianHugh** [comment]
No worries, excited to see the progress!
